libtelemetry/src/main/java/com/mapbox/android/telemetry/TelemetryUtils.java

```diff
@@ -21,6 +26,118 @@
   private static final String THREE_STRING_FORMAT = "%s/%s/%s";
   private static final SimpleDateFormat dateFormat = new SimpleDateFormat(DATE_AND_TIME_PATTERN, Locale.US);
   private static final Locale DEFAULT_LOCALE = Locale.US;
+  private static final int UNAVAILABLE_BATTERY_LEVEL = 100;
+  static TelephonyManager telephonyManager = null;
+  static ActivityManager activityManager = null;
+
+
+  public static String toHumanReadableAscii(String s) {
+    for (int i = 0, length = s.length(), c; i < length; i += Character.charCount(c)) {
+      c = s.codePointAt(i);
+      if (c > '\u001f' && c < '\u007f') {
+        continue;
+      }
+
+      Buffer buffer = new Buffer();
+      buffer.writeUtf8(s, 0, i);
+      for (int j = i; j < length; j += Character.charCount(c)) {
+        c = s.codePointAt(j);
+        buffer.writeUtf8CodePoint(c > '\u001f' && c < '\u007f' ? c : '?');
+      }
+      return buffer.readUtf8();
+    }
+    return s;
+  }
+
+  static String getApplicationState() {
+    if (activityManager == null) {
+      activityManager = (ActivityManager) MapboxTelemetry.applicationContext.getSystemService(Context.ACTIVITY_SERVICE);
+    }
+
+    List<ActivityManager.RunningAppProcessInfo> appProcesses = activityManager.getRunningAppProcesses();
+    if (appProcesses == null) {
+      return "";
+    }
+
+    String packageName = MapboxTelemetry.applicationContext.getPackageName();
+    for (ActivityManager.RunningAppProcessInfo appProcess : appProcesses) {
+      if (appProcess.importance == ActivityManager.RunningAppProcessInfo.IMPORTANCE_FOREGROUND
+        && appProcess.processName.equals(packageName)) {
+        return "Foreground";
+      }
+    }
+
+    return "Background";
+  }
+
+  static int getBatteryLevel() {
+    Intent batteryStatus = MapboxTelemetry.batteryStatus;
+
+    if (batteryStatus != null) {
+      int level = batteryStatus.getIntExtra(BatteryManager.EXTRA_LEVEL, -1);
+      int scale = batteryStatus.getIntExtra(BatteryManager.EXTRA_SCALE, -1);
+      return Math.round((level / (float) scale) * 100);
+    } else {
+      return UNAVAILABLE_BATTERY_LEVEL;
+    }
+  }
+
+  static boolean isPluggedIn() {
+    Intent batteryStatus = MapboxTelemetry.batteryStatus;
+
+    if (batteryStatus != null) {
+      int chargePlug = batteryStatus.getIntExtra(BatteryManager.EXTRA_PLUGGED, -1);
+      if (chargePlug == BatteryManager.BATTERY_PLUGGED_USB
+        || chargePlug == BatteryManager.BATTERY_PLUGGED_AC) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  static String getCellularNetworkType() {
```

Comment:
Why not using the already refactored and discussed code from `MapEventFactory#obtainCellularNetworkType` (removed as part of this PR)?